cloudwatch_logs_common/include/cloudwatch_logs_common/log_service.h

```diff
@@ -37,6 +37,14 @@
 namespace Aws {
 namespace CloudWatchLogs {
 
+/**
+ * See https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html
+ * PutLogEvents is limited to 5 requests per second per stream.
+ * Though this implies a 200ms minimum publish period, in practice we have witnessed
```

Comment:
Why not use the status as well? https://sdk.amazonaws.com/cpp/api/0.12.9/d7/d9b/_cloud_watch_logs_errors_8h_source.html

Reply:
As long as the publisher will fail if the PutLogEvents rate limit is exceeded then we can limit elsewhere.

Reply:
It does fail as in any other failure, and keeps writing to disk, and taking incoming logs. Even resets and keeps trying to upload after a delay. Here we manage the case where we don't get the error, and continue smoothly uploading at the same rate the whole time.